pypiserver/core.py

```diff
@@ -181,7 +181,8 @@ def guess_pkgname_and_version(path):
 
 
 def normalize_pkgname(name):
-    return name.lower().replace("-", "_")
+    """Perform PEP 503 normalization"""
+    return re.sub(r"[-_.]+", "-", name).lower()
```

Comment:
pkg_resources is part of setuptools though, and I think setuptools is best left as a build-time dependency. Indeed, the wheel format is intended to more cleanly separate build-time bits from run-time.